firefox-ios/Client/Frontend/Browser/Toolbars/Redux/ToolbarState.swift

```diff
@@ -432,6 +435,31 @@ struct ToolbarState: ScreenState, Sendable, Equatable {
         )
     }
 
+    private static func handleNavigationMiddleButtonDidChange(state: Self, action: Action) -> ToolbarState {
+        guard let toolbarAction = action as? ToolbarAction else { return defaultState(from: state) }
+
+        return ToolbarState(
+            windowUUID: state.windowUUID,
+            toolbarPosition: state.toolbarPosition,
+            toolbarLayout: state.toolbarLayout,
+            isPrivateMode: state.isPrivateMode,
+            addressToolbar: AddressBarState.reducer(state.addressToolbar, toolbarAction),
+            navigationToolbar: NavigationBarState.reducer(state.navigationToolbar, toolbarAction),
+            isShowingNavigationToolbar: state.isShowingNavigationToolbar,
+            isShowingTopTabs: state.isShowingTopTabs,
+            canGoBack: state.canGoBack,
+            canGoForward: state.canGoForward,
+            numberOfTabs: state.numberOfTabs,
+            scrollAlpha: state.scrollAlpha,
+            showMenuWarningBadge: state.showMenuWarningBadge,
+            isNewTabFeatureEnabled: state.isNewTabFeatureEnabled,
+            canShowDataClearanceAction: state.canShowDataClearanceAction,
+            canShowNavigationHint: state.canShowNavigationHint,
+            shouldAnimate: state.shouldAnimate,
+            isTranslucent: state.isTranslucent
+        )
```

Comment:
If I'm understanding this correctly, it seems that we're calling this to re-trigger the navigationToolbar reducer? I see that we do that for `handleReaderModeStateChanged` as well unless I'm missing something. Maybe we can combine and make a generic method to handle both?